runtime/src/accounts_db.rs

```diff
@@ -3768,20 +3769,21 @@ impl AccountsDb {
                     let mut hash_time = Measure::start("hash_accounts");
                     let mut stats = BankHashStats::default();
                     let len = accounts_and_meta_to_store.len();
-                    let mut hashes = Vec::with_capacity(len);
+                    let mut hashes_store = Vec::with_capacity(len);
                     for account in accounts {
                         stats.update(account.1);
                         let hash = Self::hash_account(slot, account.1, account.0);
-                        hashes.push(hash);
+                        hashes_store.push(hash);
                     }
+                    let hashes_refs = hashes_store.iter().collect::<Vec<_>>();
```

Comment:
this code path is only when caching is not enabled or we're not a cached store. I'm not sure how often this happens. The additional work here is to make an array of refs to the hash value array we had to create.